acapy_agent/did/cheqd/cheqd_manager.py

```diff
@@ -1,29 +1,77 @@
 """DID manager for Cheqd."""
 
+import logging
+
 from aiohttp import web
-from .registrar import DidCheqdRegistrar
+
 from ...core.profile import Profile
+from ...resolver.base import DIDNotFound
+from ...resolver.default.cheqd import CheqdDIDResolver
 from ...wallet.base import BaseWallet
 from ...wallet.crypto import validate_seed
 from ...wallet.did_method import CHEQD, DIDMethods
 from ...wallet.did_parameters_validation import DIDParametersValidation
 from ...wallet.error import WalletError
 from ...wallet.key_type import ED25519
-from ...wallet.util import b64_to_bytes, bytes_to_b64, b58_to_bytes
+from ...wallet.util import b58_to_bytes, b64_to_bytes, bytes_to_b64
+from .registrar import DidCheqdRegistrar
+
+LOGGER = logging.getLogger(__name__)
 
 
 class DidCheqdManager:
     """DID manager for Cheqd."""
 
     registrar: DidCheqdRegistrar
+    resolver: CheqdDIDResolver
 
     def __init__(self, profile: Profile) -> None:
-        """Initialize the DID  manager."""
+        """Initialize the Cheqd DID manager."""
         self.profile = profile
         self.registrar = DidCheqdRegistrar()
+        self.resolver = CheqdDIDResolver()
+
+    async def sign_requests(wallet, signing_requests, verkey=None):
+        """Sign all requests in the siginingRequests array.
+
+        Args:
+            wallet: Wallet instance used to sign messages.
+            signing_requests: List of signing request dictionaries.
+            verkey (optional): Pre-determined verkey to use for signing. \
+                If None, retrieve verkey from the wallet.
+
+        Returns:
+            List of signed responses, each containing 'kid' and 'signature'.
+        """
+        signed_responses = []
+        for sign_req in signing_requests:
+            kid = sign_req.get("kid")
+            payload_to_sign = sign_req.get("serializedPayload")
+            if verkey:
+                # assign verkey to kid in wallet
+                await wallet.assign_kid_to_key(verkey, kid)
```

Comment:
FYI I'm going to make a change here, s.t only if a kid is not mapped with a key then I'm allowing to assign it. This can prevent incorrect assigning if the request is invalid